aten/src/ATen/core/LazyTensor.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <c10/core/TensorImpl.h>
+#include <ATen/core/ivalue.h>
+
+namespace at {
+
+struct CAFFE2_API LazyTensorImpl : public TensorImpl {
```

Comment:
In this struct, I don't see any place for storing materialized results after you run `to_eager` on a graph. This seems to imply to me that if I do something like:

```
x = ... some lazy computation ...
x.to_eager()
x.to_eager()
```

I will end up redoing the entirety of lazy computation twice. Is that true?

Reply:
yep, that's true